cmd/notation/internal/plugin/plugin.go

```diff
@@ -52,35 +56,54 @@ const (
 // URL
 const DownloadPluginFromURLTimeout = 10 * time.Minute
 
-// DownloadPluginFromURL downloads plugin source from url to a tmp file on file
-// system
-func DownloadPluginFromURL(ctx context.Context, pluginURL string, tmpFile io.Writer) error {
+// DownloadPluginFromURL downloads plugin source from url to a tmp dir on file
+// system. On success, it returns the downloaded file.
+func DownloadPluginFromURL(ctx context.Context, pluginURL, tmpDir string) (*os.File, error) {
```

Comment:
The `*os.File` has been closed, so we cannot read or write it anymore. Looks like we only need the file path. Can we only return the path instead of a closed `*os.File`?